common/src/api/external/mod.rs

```diff
@@ -2980,6 +2961,33 @@ pub struct SwitchPortAddressConfig {
     pub interface_name: String,
 }
 
+/// An IP address configuration for a port settings object.
+#[derive(Clone, Debug, Deserialize, JsonSchema, Serialize, PartialEq)]
+pub struct SwitchPortAddressView {
```

Comment:
I see this matches up with `SwitchPortSettingsView`, but we don't usually put `View` on the end of these structs we call views. In this case I think `SwitchPortAddress` might be ok, though it makes it sound like it's just the address. But either nothing or something more descriptive than `View` would be an improvement.

`SwitchPortSettingsView` can't be changed to `SwitchPortSettings` because there is already a `SwitchPortSettings`, and `SwitchPortSettingsConfig` is ridiculous. What if instead of the `settings` key, you took that `#[serde(flatten)] identity` bit and stuck it directly in `SwitchPortSettingsView`? Then you could rename the struct `SwitchPortSettings`.

Reply:
Giving this a try now

Reply:
It looks like the existing `SwitchPortSettings` struct that wraps the `IdentityMetadata` is what we return when we list the `SwitchPortSettings` resource. I'm not sure on the original motivation for this, but the complete `SwitchPortSettingsView` struct is quite large and requires a bit of work to populate, so maybe we didn't want to do that when listing the resource.

Would `SwitchPortSettingsIdentity` make more sense for the wrapper struct? That would allow us to use `SwitchPortSettings` for the "view" struct.

Reply:
I've updated these type names to `SwitchPortSettings` and `SwitchPortSettingsIdentity`.

Reply:
Cool, I think it looks good.

Reply:
👍